src/device.rs

```diff
@@ -341,6 +342,76 @@ impl Device {
         t.submit::<ControlOut>(data);
         TransferFuture::new(t)
     }
+
+    /// Get the device descriptor.
+    ///
+    /// The only situation when it returns `None` is
+    /// that the cached descriptors contain no valid device descriptor.
+    ///
+    /// ### Platform-specific notes
+    ///
+    /// * This is only supported on Linux.
+    /// * On Linux, this method uses descriptors cached in memory, instead
+    ///   of sending a request to the device for a descriptor.
+    #[cfg(any(target_os = "linux", target_os = "android"))]
+    pub fn get_device_descriptor(&self) -> Option<DeviceDescriptor> {
```

Comment:
```suggestion
    pub fn device_descriptor(&self) -> Option<DeviceDescriptor> {
```

Per Rust API guidelines, since it's a getter for an internal field and not making a request to the device.